### ISSUE TITLE
Wallet using an outpoint that doesn't exist onchain

### ISSUE THREAD

**Issue by Christewart** [bitcoin-s/bitcoin-s#4008]
Running on mainnet neutrino sqlite3 database on  52dcf51e82f4cc06cc8eff1c82c835ed68910551

The second tx sent (9544ef874b016f23229d2a44b22302deab92b1265b9262ed8675e25beee210f5) does not appear in an explorer, and we get a weird `WARN` log around inventories that we don't have?

It appears that the bitcoin node we are connected to requests the outpoint we use to fund `9544ef...10f5`. The outpoint we use to fund this transaction is [`c68e9fc3075e6ca7bcee17f3d64cd868987b68ce406846add3a722f1059da0b3:1`](https://mempool.space/tx/c68e9fc3075e6ca7bcee17f3d64cd868987b68ce406846add3a722f1059da0b3).

```
2022-01-24T15:54:17UTC INFO [DLCWallet$DLCWalletImpl] Sending 0.00100000 BTC to bc1q3hr5n57nhlcj6vcj5pkuay4f6vy2fnwn7uv23v at feerate 1 sats/vbyte
2022-01-24T15:54:17UTC INFO [DLCWallet$DLCWalletImpl] Attempting to fund a tx for amt=0.00100000 BTC with feeRate=1 sats/vbyte
2022-01-24T15:54:17UTC INFO [DLCWallet$DLCWalletImpl] Spending UTXOs: Vector(TransactionOutPoint(c68e9fc3075e6ca7bcee17f3d64cd868987b68ce406846add3a722f1059da0b3:1) state=ConfirmedReceived)
2022-01-24T15:54:17UTC INFO [DLCWallet$DLCWalletImpl] UTXO 0 details: TransactionOutput(807273 sats,wpkh(e0dc8b78db5987c3c5074e90d29b7f99a2b582ee))
2022-01-24T15:54:17UTC INFO [DLCWallet$DLCWalletImpl] Processing TX from our wallet, transaction=9544ef874b016f23229d2a44b22302deab92b1265b9262ed8675e25beee210f5 with blockHash=None
2022-01-24T15:54:17UTC INFO [DLCWallet$DLCWalletImpl] Adding UTXO to wallet: 9544ef874b016f23229d2a44b22302deab92b1265b9262ed8675e25beee210f5:1 amt=707132 sats
2022-01-24T15:54:17UTC INFO [DLCWallet$DLCWalletImpl] Successfully inserted UTXO 9544ef874b016f23229d2a44b22302deab92b1265b9262ed8675e25beee210f5:1 into DB
2022-01-24T15:54:17UTC INFO [DLCWallet$DLCWalletImpl] Finished processing 1 received outputs, it took=24ms
2022-01-24T15:54:17UTC INFO [DLCWallet$DLCWalletImpl] Finished processing 1 spent outputs, it took=11ms
2022-01-24T15:54:17UTC INFO [DLCWallet$DLCWalletImpl] Processing of internal transaction=9544ef874b016f23229d2a44b22302deab92b1265b9262ed8675e25beee210f5 resulted in changeOutputs=1 and spentUTXOs=1
2022-01-24T15:54:17UTC INFO [NeutrinoNode] Sending out tx message for tx=Vector(9544ef874b016f23229d2a44b22302deab92b1265b9262ed8675e25beee210f5)
2022-01-24T15:54:17UTC INFO [DLCWallet$DLCWalletImpl] Finished processing 1 received outputs, it took=2ms
2022-01-24T15:54:17UTC INFO [DLCWallet$DLCWalletImpl] Finished processing 1 spent outputs, it took=19ms
2022-01-24T15:54:21UTC INFO [DataMessageHandler] Received a getdata message for inventories=List(Inventory(MsgWitnessTx,DoubleSha256Digest(f510e2ee5be27586ed62925b26b192abde0223b2442a9d22236f014b87ef4495)))
2022-01-24T15:54:25UTC INFO [DataMessageHandler] Received a getdata message for inventories=List(Inventory(MsgWitnessTx,DoubleSha256Digest(b3a09d05f122a7d3ad466840ce687b9868d84cd6f317eebca76c5e07c39f8ec6)))
2022-01-24T15:54:25UTC WARN [DataMessageHandler] Got request to send data with hash=DoubleSha256Digest(b3a09d05f122a7d3ad466840ce687b9868d84cd6f317eebca76c5e07c39f8ec6), but found nothing
```

![Screenshot from 2022-01-24 09-56-59](https://user-images.githubusercontent.com/3514957/150817656-67a5e79b-0642-4d2c-8a9d-0a623d55f9c8.png)

Here is the decoded transaction that failed to broadcast `9544ef874b016f23229d2a44b22302deab92b1265b9262ed8675e25beee210f5`

```
{
  "txid": "9544ef874b016f23229d2a44b22302deab92b1265b9262ed8675e25beee210f5",
  "wtxid": "f934c37c9e4a009ec973dbd64e39739b07a55100c00a2c082a102bcada414f08",
  "version": 2,
  "size": 222,
  "vsize": 141,
  "weight": 561,
  "locktime": 0,
  "vin": [
    {
      "txid": "c68e9fc3075e6ca7bcee17f3d64cd868987b68ce406846add3a722f1059da0b3",
      "hex": "b3a09d05f122a7d3ad466840ce687b9868d84cd6f317eebca76c5e07c39f8ec6010000000000000000",
      "vout": 1,
      "scriptSig": [
        
      ],
      "txinwitness": {
        "hex": "02473044022070ea44e5f0ea4f6e3c515d6476f3afdc0b0ae3bc366fcd8128fee8a968b247450220433f97b52a827504c70265170c3aedd61bfadca4b1a6d03029b80713fd2760f601210334e15179a23e66a9dc0ea10f1db2666d44a6618b7344dd76c88c86cf7ad665f7",
        "scriptType": "P2WPKH",
        "pubKey": "0334e15179a23e66a9dc0ea10f1db2666d44a6618b7344dd76c88c86cf7ad665f7",
        "signature": "3044022070ea44e5f0ea4f6e3c515d6476f3afdc0b0ae3bc366fcd8128fee8a968b247450220433f97b52a827504c70265170c3aedd61bfadca4b1a6d03029b80713fd2760f601"
      },
      "sequence": 0
    }
  ],
  "vout": [
    {
      "value": 100000,
      "n": 0,
      "scriptPubKey": [
        "OP_0",
        "BytesToPushOntoStackImpl(20)",
        "ByteVector(20 bytes, 0x8dc749d3d3bff12d3312a06dce92a9d308a4cdd3)"
      ],
      "hex": "a0860100000000001600148dc749d3d3bff12d3312a06dce92a9d308a4cdd3"
    },
    {
      "value": 707132,
      "n": 1,
      "scriptPubKey": [
        "OP_0",
        "BytesToPushOntoStackImpl(20)",
        "ByteVector(20 bytes, 0x43ce14fd56860f99a2a988659fd63b6cfcdf3197)"
      ],
      "hex": "3cca0a000000000016001443ce14fd56860f99a2a988659fd63b6cfcdf3197"
    }
  ]
}
```

Here is [`c68e9fc3075e6ca7bcee17f3d64cd868987b68ce406846add3a722f1059da0b3`](https://mempool.space/tx/c68e9fc3075e6ca7bcee17f3d64cd868987b68ce406846add3a722f1059da0b3) which is the parent transaction to `9544ef874b016f23229d2a44b22302deab92b1265b9262ed8675e25beee210f5`

```
{
  "txid": "c68e9fc3075e6ca7bcee17f3d64cd868987b68ce406846add3a722f1059da0b3",
  "wtxid": "c1678dc7a88074652ef3ed15160deff70a631d80b26e9ca0bf5fc9d04d99a1e0",
  "version": 2,
  "size": 518,
  "vsize": 276,
  "weight": 1103,
  "locktime": 0,
  "vin": [
    {
      "txid": "9ca2f15bd4032445761ca3a6a215ca7b70383d650dc0643cf7792f3a127baa06",
      "hex": "06aa7b123a2f79f73c64c00d653d38707bca15a2a6a31c76452403d45bf1a29c000000000000000000",
      "vout": 0,
      "scriptSig": [
        
      ],
      "txinwitness": {
        "hex": "024730440220568b31c6139a39b61c6fa39cf6969a2b9bf4aa06c9de9cb2b64d169c5fb9be6502205efd7b74018a35c9cf7d3f221e1d130c4be3220512876733dbd5672305058c59012102ea565e7a6faa7044c48cf692782982b3bdbd1091af888a375cecf1b73052c743",
        "scriptType": "P2WPKH",
        "pubKey": "02ea565e7a6faa7044c48cf692782982b3bdbd1091af888a375cecf1b73052c743",
        "signature": "30440220568b31c6139a39b61c6fa39cf6969a2b9bf4aa06c9de9cb2b64d169c5fb9be6502205efd7b74018a35c9cf7d3f221e1d130c4be3220512876733dbd5672305058c5901"
      },
      "sequence": 0
    },
    {
      "txid": "c23e3c9982f3cd1776c9a1269ed5d858324d2e41bddb32b152c97d7d30527ae0",
      "hex": "e07a52307d7dc952b132dbbd412e4d3258d8d59e26a1c97617cdf382993c3ec2000000000000000000",
      "vout": 0,
      "scriptSig": [
        
      ],
      "txinwitness": {
        "hex": "0247304402202aadf171fe1f1edcb4edae9f0a3724ba26b5c7861a149c03ec4be76a663b780102203340682af0566025708e26a3224cc6915c42c7e523b66541aa1da84d5b212c350121036fd1461ac9dc42d4d4ae556197a981e4da281a42d2c8e8cbf0cb5f3e3224fe96",
        "scriptType": "P2WPKH",
        "pubKey": "036fd1461ac9dc42d4d4ae556197a981e4da281a42d2c8e8cbf0cb5f3e3224fe96",
        "signature": "304402202aadf171fe1f1edcb4edae9f0a3724ba26b5c7861a149c03ec4be76a663b780102203340682af0566025708e26a3224cc6915c42c7e523b66541aa1da84d5b212c3501"
      },
      "sequence": 0
    },
    {
      "txid": "a950c9b6fddfb4395f806cfebf1997251169abb1870bd61f4a1532e7c5e2f8e4",
      "hex": "e4f8e2c5e732154a1fd60b87b1ab6911259719bffe6c805f39b4dffdb6c950a9000000000000000000",
      "vout": 0,
      "scriptSig": [
        
      ],
      "txinwitness": {
        "hex": "02473044022054aa5222a6f65f76cf818bc7a23a649947cd5e32c72c1d34ce9f75b8a980d1db02203a4954c73d858185a5ea60002f1a471fd6661c1903760f494c9611f81bbfab400121037a0ea60f52ae0f307aa6ea2e3f7b8d28e993f0093fc5a15630a9d22973b0c8a4",
        "scriptType": "P2WPKH",
        "pubKey": "037a0ea60f52ae0f307aa6ea2e3f7b8d28e993f0093fc5a15630a9d22973b0c8a4",
        "signature": "3044022054aa5222a6f65f76cf818bc7a23a649947cd5e32c72c1d34ce9f75b8a980d1db02203a4954c73d858185a5ea60002f1a471fd6661c1903760f494c9611f81bbfab4001"
      },
      "sequence": 0
    }
  ],
  "vout": [
    {
      "value": 1000000,
      "n": 0,
      "scriptPubKey": [
        "OP_0",
        "BytesToPushOntoStackImpl(20)",
        "ByteVector(20 bytes, 0x6d1edfd9a3f14b58c2056040309bf79269077074)"
      ],
      "hex": "40420f00000000001600146d1edfd9a3f14b58c2056040309bf79269077074"
    },
    {
      "value": 807273,
      "n": 1,
      "scriptPubKey": [
        "OP_0",
        "BytesToPushOntoStackImpl(20)",
        "ByteVector(20 bytes, 0xe0dc8b78db5987c3c5074e90d29b7f99a2b582ee)"
      ],
      "hex": "69510c0000000000160014e0dc8b78db5987c3c5074e90d29b7f99a2b582ee"
    }
  ]
}

```

The output [c68e9fc3075e6ca7bcee17f3d64cd868987b68ce406846add3a722f1059da0b3:1](https://mempool.space/tx/c68e9fc3075e6ca7bcee17f3d64cd868987b68ce406846add3a722f1059da0b3) is spent and for some reason our wallet did not see it.

**Comment by Christewart** [comment]
I sent a third transaction out of the wallet and it seemed to have broadcast fine

https://mempool.space/tx/86e57861d418747aff7fa95536a628104200fbefcaef97b7936c92d68fe13bc3

```
[info] 2022-01-24T15:54:25UTC WARN [DataMessageHandler] Got request to send data with hash=DoubleSha256Digest(b3a09d05f122a7d3ad466840ce687b9868d84cd6f317eebca76c5e07c39f8ec6), but found nothing
[info] 2022-01-24T16:02:37UTC INFO [DataMessageHandler] Received headers message with 1 headers
[info] 2022-01-24T16:02:37UTC INFO [DataMessageHandler] Starting to fetch filter headers in data message handler
[info] 2022-01-24T16:02:38UTC INFO [DataMessageHandler] Got 1 compact filter header hashes
[info] 2022-01-24T16:02:38UTC INFO [DataMessageHandler] Done syncing filter headers, beginning to sync filters in datamessagehandler
[info] 2022-01-24T16:02:38UTC INFO [PeerMessageSender] Requesting compact filters from FilterSyncMarker(startHeight=720204, stopBlockHash=000000000000000000060314724df714d341d469544bb935652014d4d0dc34fd)
[info] 2022-01-24T16:02:38UTC INFO [DataMessageHandler] We are synced
[info] 2022-01-24T16:02:38UTC INFO [DataMessageHandler] Processing 1 filters
[info] 2022-01-24T16:04:05UTC INFO [DataMessageHandler] Received headers message with 1 headers
[info] 2022-01-24T16:04:05UTC INFO [DataMessageHandler] Starting to fetch filter headers in data message handler
[info] 2022-01-24T16:04:05UTC INFO [DataMessageHandler] Got 1 compact filter header hashes
[info] 2022-01-24T16:04:05UTC INFO [DataMessageHandler] Done syncing filter headers, beginning to sync filters in datamessagehandler
[info] 2022-01-24T16:04:05UTC INFO [PeerMessageSender] Requesting compact filters from FilterSyncMarker(startHeight=720205, stopBlockHash=000000000000000000054168b18fa540357880ce23f9c005453e6dbd2d58b31e)
[info] 2022-01-24T16:04:05UTC INFO [DataMessageHandler] We are synced
[info] 2022-01-24T16:04:05UTC INFO [DataMessageHandler] Processing 1 filters
[info] 2022-01-24T16:07:21UTC INFO [DataMessageHandler] Received headers message with 1 headers
[info] 2022-01-24T16:07:21UTC INFO [DataMessageHandler] Starting to fetch filter headers in data message handler
[info] 2022-01-24T16:07:21UTC INFO [DataMessageHandler] Got 1 compact filter header hashes
[info] 2022-01-24T16:07:21UTC INFO [DataMessageHandler] Done syncing filter headers, beginning to sync filters in datamessagehandler
[info] 2022-01-24T16:07:21UTC INFO [PeerMessageSender] Requesting compact filters from FilterSyncMarker(startHeight=720206, stopBlockHash=00000000000000000003d6ce1c4b0514cce7c7d2aab9df292a4b43815090bb99)
[info] 2022-01-24T16:07:22UTC INFO [DataMessageHandler] We are synced
[info] 2022-01-24T16:07:22UTC INFO [DataMessageHandler] Processing 1 filters
[info] 2022-01-24T16:11:00UTC INFO [DataMessageHandler] Received headers message with 1 headers
[info] 2022-01-24T16:11:00UTC INFO [DataMessageHandler] Starting to fetch filter headers in data message handler
[info] 2022-01-24T16:11:00UTC INFO [DataMessageHandler] Got 1 compact filter header hashes
[info] 2022-01-24T16:11:00UTC INFO [DataMessageHandler] Done syncing filter headers, beginning to sync filters in datamessagehandler
[info] 2022-01-24T16:11:00UTC INFO [PeerMessageSender] Requesting compact filters from FilterSyncMarker(startHeight=720207, stopBlockHash=000000000000000000035ff1d73f55b7ee77d266df3adf869e48070cff37ef53)
[info] 2022-01-24T16:11:01UTC INFO [DataMessageHandler] We are synced
[info] 2022-01-24T16:11:01UTC INFO [DataMessageHandler] Processing 1 filters
[info] 2022-01-24T16:11:46UTC INFO [DLCWallet$DLCWalletImpl] Sending 0.00100000 BTC to bc1q3hr5n57nhlcj6vcj5pkuay4f6vy2fnwn7uv23v at feerate 1 sats/vbyte
[info] 2022-01-24T16:11:46UTC INFO [DLCWallet$DLCWalletImpl] Attempting to fund a tx for amt=0.00100000 BTC with feeRate=1 sats/vbyte
[info] 2022-01-24T16:11:46UTC INFO [DLCWallet$DLCWalletImpl] Spending UTXOs: Vector(TransactionOutPoint(be73ff50dcb2926b9840c249541612f59bf6130577ddd7ca58d41f8398709778:0) state=ConfirmedReceived)
[info] 2022-01-24T16:11:46UTC INFO [DLCWallet$DLCWalletImpl] UTXO 0 details: TransactionOutput(46172840 sats,wpkh(5ec1a6d98101f4a59b2431b30e91144991153f9c))
[info] 2022-01-24T16:11:46UTC INFO [DLCWallet$DLCWalletImpl] Processing TX from our wallet, transaction=86e57861d418747aff7fa95536a628104200fbefcaef97b7936c92d68fe13bc3 with blockHash=None
[info] 2022-01-24T16:11:46UTC INFO [DLCWallet$DLCWalletImpl] Adding UTXO to wallet: 86e57861d418747aff7fa95536a628104200fbefcaef97b7936c92d68fe13bc3:0 amt=46072699 sats
[info] 2022-01-24T16:11:46UTC INFO [DLCWallet$DLCWalletImpl] Successfully inserted UTXO 86e57861d418747aff7fa95536a628104200fbefcaef97b7936c92d68fe13bc3:0 into DB
[info] 2022-01-24T16:11:46UTC INFO [DLCWallet$DLCWalletImpl] Finished processing 1 received outputs, it took=23ms
[info] 2022-01-24T16:11:46UTC INFO [DLCWallet$DLCWalletImpl] Finished processing 1 spent outputs, it took=13ms
[info] 2022-01-24T16:11:46UTC INFO [DLCWallet$DLCWalletImpl] Processing of internal transaction=86e57861d418747aff7fa95536a628104200fbefcaef97b7936c92d68fe13bc3 resulted in changeOutputs=1 and spentUTXOs=1
[info] 2022-01-24T16:11:46UTC INFO [NeutrinoNode] Sending out tx message for tx=Vector(86e57861d418747aff7fa95536a628104200fbefcaef97b7936c92d68fe13bc3)
[info] 2022-01-24T16:11:46UTC INFO [DLCWallet$DLCWalletImpl] Finished processing 1 received outputs, it took=2ms
[info] 2022-01-24T16:11:46UTC INFO [DLCWallet$DLCWalletImpl] Finished processing 1 spent outputs, it took=16ms
[info] 2022-01-24T16:11:50UTC INFO [DataMessageHandler] Received a getdata message for inventories=List(Inventory(MsgWitnessTx,DoubleSha256Digest(c33be18fd6926c93b797efcaeffb00421028a63655a97fff7a7418d46178e586)))
```

note the timestamps to the right on `tmux`

![Screenshot from 2022-01-24 10-13-43](https://user-images.githubusercontent.com/3514957/150820665-a9f5819d-6921-4230-9b54-663fe88a0026.png)

**Comment by Christewart** [comment]
fixed in #4290 / #4296
